subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/coap.h>
+#include <zephyr/net/coap_client.h>
+#if defined(CONFIG_POSIX_API)
+#include <zephyr/posix/arpa/inet.h>
+#include <zephyr/posix/netdb.h>
+#include <zephyr/posix/unistd.h>
+#include <zephyr/posix/poll.h>
+#include <zephyr/posix/sys/socket.h>
+#else
+#include <zephyr/net/socket.h>
+#endif
+#include <modem/lte_lc.h>
+#include <zephyr/random/rand32.h>
+#include <nrf_socket.h>
+#include <nrf_modem_at.h>
+#include <date_time.h>
+#include <net/nrf_cloud.h>
+#include <net/nrf_cloud_coap.h>
+#include <cJSON.h>
+#include <version.h>
+#include "nrf_cloud_codec_internal.h"
+#include "dtls.h"
+#include "coap_codec.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(nrf_cloud_coap_transport, CONFIG_NRF_CLOUD_COAP_LOG_LEVEL);
+
+/** @TODO: figure out whether to make this a Kconfig value or place in a header */
+#define CDDL_VERSION "1"
+#define MAX_COAP_PATH 256
+#define MAX_RETRIES 10
+#define JWT_BUF_SZ 700
+#define VER_STRING_FMT "mver=%s&cver=%s&dver=%s"
+#define VER_STRING_FMT2 "cver=" CDDL_VERSION "&dver=" BUILD_VERSION_STR
+#define BUILD_VERSION_STR STRINGIFY(BUILD_VERSION)
+
+static struct sockaddr_storage server;
+
+static int sock;
+static bool authorized;
+
+static struct connection_info
+{
+	uint8_t s4_addr[4];
+	uint8_t d4_addr[4];
+} connection_info;
+
+static struct coap_client coap_client;
+static int nrf_cloud_coap_authorize(void);
+
+bool nrf_cloud_coap_is_connected(void)
+{
+	if (!authorized) {
+		LOG_ERR("Not connected and authorized");
```

Comment:
This error trace doesn't look nice when an application is using this function to check the state. This function does not know whether not being authorized is an error or not.